### DISCUSSION TITLE
Is there a replacement for UnusedParameterCheck? I'm sad its gone.

### DISCUSSION THREAD

**Discussion by sara-jv** [checkstyle/checkstyle#12598] [Q&A] [answered]
http://api.dpml.net/checkstyle/3.5/com/puppycrawl/tools/checkstyle/checks/usage/UnusedParameterCheck.html

Looking for something similar in the latest version. Can anyone recommend me something?

Testing for this use case: 

... main method .. 

        test("hello");
        return Longs.tryParse(request.getCategoryId());
    }

    @Nullable
    Long test(String unused) {
        System.out.println("Hello World");
        return null;
    }

I want the build to fail

**Comment by romani** [comment]
I can not find anything on this Check , looks like it was dropped before checkstyle version 5. 

You can create issue to create such Check.

**Reply by rnveach** [reply, marked as answer]
Check was moved to https://github.com/checkstyle/contribution/blob/master/usage/src/checkstyle/com/puppycrawl/tools/checkstyle/checks/usage/UnusedParameterCheck.java in 2015 (Checkstyle 6.4) and before that it was most likely disabled.

I will warn you it has not been maintained and I have no clue what state it is in so it is not likely to compile with current versions of Checkstyle as is. You can use it but you will probably be on your own including running it. I doubt anyone will take time for it even to bring it back if possible.

**Reply by romani** [reply]
Yes, https://github.com/checkstyle/contribution/commits/master/usage/src/checkstyle/com/puppycrawl/tools/checkstyle/checks/usage/UnusedParameterCheck.java

This module was not in main set of Checks, may be never part of our jar so I moved it out of main repo at that time to make it very clear it is not ready for usage.

Request to have such module is very valid, we can create issue to make it.

**Reply by romani** [reply]
oldest version in archive we have is https://checkstyle.sourceforge.io/version/4.4/
it was not existed in availbale Checks even on 4.4 - https://checkstyle.sourceforge.io/version/4.4/availablechecks.html

all available versions in archive as of now:
```
10.0
10.1
10.2
10.3
10.3.1
10.3.2
10.3.3
10.3.4
10.4
10.5.0
4.4
5.5
5.6
5.7
5.8
5.9
6.0
6.1
6.1.1
6.10
6.10.1
6.11
6.11.1
6.11.2
6.12
6.12.1
6.13
6.14
6.15
6.17
6.18
6.19
6.2
6.3
6.4.1
6.5
6.6
6.7
6.8
6.8.1
6.8.2
6.9
7.0
7.1
7.1.1
7.1.2
7.2
7.3
7.4
7.5
7.5.1
7.6
7.7
7.8
7.8.1
7.8.2
8.0
8.1
8.10
8.10.1
8.11
8.12
8.13
8.14
8.15
8.16
8.17
8.18
8.19
8.2
8.20
8.21
8.22
8.23
8.24
8.25
8.26
8.27
8.28
8.29
8.3
8.30
8.31
8.32
8.33
8.34
8.35
8.36
8.36.1
8.36.2
8.37
8.38
8.39
8.4
8.40
8.41
8.41.1
8.42
8.43
8.44
8.45
8.45.1
8.5
8.7
8.8
8.9
9.0
9.0.1
9.1
9.2
9.2.1
9.3
```

**Reply by rnveach** [reply]
> all available versions in archive as of now

The following versions are missing from archive:
````
5.0
5.1
5.2
5.3
5.4
6.14.1
6.16
6.16.1
6.4
7.6.1
8.6
````

**Reply by romani** [reply]
yes, there might be some deploy problems, yes we do not have files for them.
I donot think we need to restore them (we can checkout to such tags and build site from sources) they are too old to be considered, if badly needed, look at documentation of sibling version
